front/admin_ui/requirements.txt

```diff
@@ -1,5 +1,7 @@
-gradio~=3.18.0
+gradio==3.28.1
+libcommon @ git+https://github.com/huggingface/datasets-server@main#subdirectory=libs/libcommon
```

Comment:
I think it will raise an error when deploying to Spaces. On my laptop it I get
```
ERROR: Package 'libcommon' requires a different Python: 3.10.11 not in '==3.9.15'
```
maybe we can have an endpoint that returns the processing graph as a JSON object, and the UI can query it with a GET request ? And this way it shows the actual processing graph of the server at the state at which is it deployed, not the one on 'main'

Reply:
> maybe we can have an endpoint that returns the processing graph as a JSON object, and the UI can query it with a GET request ? And this way it shows the actual processing graph of the server at the state at which is it deployed, not the one on 'main'

Another way would be to update the space when we deploy datasets-server to prod.
Maybe we could split libcommon into two libs, if it helps reducing the number of dependencies in the space.

Reply:
> Another way would be to update the space when we deploy datasets-server to prod.

How to do this correctly? Currently `libcommon` dependency is provided as `libcommon @ git+https://github.com/huggingface/datasets-server@main#subdirectory=libs/libcommon`, as far as I understand this is not equivalent to what is deployed to prod? 

> Maybe we could split libcommon into two libs, if it helps reducing the number of dependencies in the space.

On my side the container is being build pretty fast anyway, so maybe the number of dependencies for the space is not that crucial

Reply:
> How to do this correctly? According to this change libcommon dependency in the space will be provided as libcommon @ git+https://github.com/huggingface/datasets-server@main#subdirectory=libs/libcommon, as far as I understand this is not equivalent to what is deployed to prod?

OK, I had not understood the issue. Not sure how we can manage this.

Reply:
Ah but if we update the space only when we deploy, by installing from `main` we will get exactly the version on the prod, right? At least most likely :D if nothing was merged to main meanwhile. But that means we must not forget about it each time.

Reply:
Yes, I think it's ok for now to do so